egs/avspeech/model.py

```diff
@@ -0,0 +1,381 @@
+        self.conv1 = nn.Conv2d(512,256,kernel_size = (7,1),padding = self.get_padding((7,1),(1,1)), dilation = (1,1))
+
+        self.conv2 = nn.Conv2d(256,256,kernel_size = (5,1),padding = self.get_padding((5,1),(1,1)), dilation = (1,1))
+
+        self.conv3 = nn.Conv2d(256,256,kernel_size = (5,1),padding = self.get_padding((5,1),(2,1)), dilation = (2,1))
+
+        self.conv4 = nn.Conv2d(256,256,kernel_size = (5,1),padding = self.get_padding((5,1),(4,1)), dilation = (4,1))
+
+        self.conv5 = nn.Conv2d(256,256,kernel_size = (5,1),padding = self.get_padding((5,1),(8,1)), dilation = (8,1))
+
+        self.conv6 = nn.Conv2d(256,256,kernel_size = (5,1),padding = self.get_padding((5,1),(16,1)), dilation = (16,1))
+
+        #Batch normalization layers
+
+        self.batch_norm1 = nn.BatchNorm2d(256)
+        self.batch_norm2 = nn.BatchNorm2d(256)
+        self.batch_norm3 = nn.BatchNorm2d(256)
+        self.batch_norm4 = nn.BatchNorm2d(256)
+        self.batch_norm5 = nn.BatchNorm2d(256)
+        self.batch_norm6 = nn.BatchNorm2d(256)
+
+    def get_padding(self,kernel_size,dilation):
+        padding = (((dilation[0])*(kernel_size[0]-1))//2,((dilation[1])*(kernel_size[1]-1))//2)
+        return padding
+
+    def forward(self,input_video):
+        # input video will be (512,75,1)
+        if len(input_video.shape) == 3:
+            input_video = input_video.unsqueeze(1)
+        #input_video = torch.transpose(input_video,1,3) # (1,75,512)
+        #print (input_video.shape)
+        #input_video = self.linear_for_512_to_1024(input_video) # (1,75,1024)
+
+        input_video = torch.transpose(input_video,1,3) # (1024,75,1)
+
+
+        output_layer = F.relu(self.batch_norm1(self.conv1(input_video)))
+        #print (output_layer.shape)
+        output_layer = F.relu(self.batch_norm2(self.conv2(output_layer)))
+        #print (output_layer.shape)
+        output_layer = F.relu(self.batch_norm3(self.conv3(output_layer)))
+        #print (output_layer.shape)
+        output_layer = F.relu(self.batch_norm4(self.conv4(output_layer)))
+        #print (output_layer.shape)
+        output_layer = F.relu(self.batch_norm5(self.conv5(output_layer)))
+        #print (output_layer.shape)
+        output_layer = F.relu(self.batch_norm6(self.conv6(output_layer)))
+        #print (output_layer.shape)
+
+        #for upsampling , as mentioned in paper
+        output_layer = nn.functional.interpolate(output_layer,size=(298,1),mode="nearest")
+        #print (output_layer.shape)
+
+        return output_layer
+
+# so now , video_output is (N,256,298,1)
+# and audio_output is  (N,8*257,298,1)
+# where N = batch_size
+
+class Audio_Visual_Fusion(nn.Module):
+
+    def __init__(self,num_person=2,device=None):
+        if isinstance(device, str):
+            device = torch.device(device)
+
+        self.device = device
+        super(Audio_Visual_Fusion, self).__init__()
+        self.num_person = num_person
+        self.input_dim = 8*257 + 256*self.num_person
```

Comment:
Do you want to try to get rid of the constants and depend on an `input_dim` param that can be computed from the STFT?